tests/integration-tests/tests/scaling/test_mpi.py

```diff
@@ -69,7 +68,6 @@ def test_mpi_ssh(scheduler, os, pcluster_config_reader, clusters_factory, test_d
     _test_mpi_ssh(remote_command_executor, scheduler, os, test_datadir)
 
 
-@timeout(10, use_signals=False, timeout_exception=RemoteCommandExecutionError)
```

Comment:
The two calls to `run_remote_script` might get stuck in case of errors due to the nature of the test. I'm ok with removing this annotation but can we add an optional timeout parameter to `run_remote_command` and `run_remote_command` functions and have this timeout propagated to the run() call? So that in this test function only when we invoke the run_remote_script we also pass `timeout=10`

Here is the doc for the run command which seems to support a timeout argument: http://docs.pyinvoke.org/en/latest/api/runners.html#invoke.runners.Runner.run

Reply:
Added in latest revision